datadog_checks_base/datadog_checks/base/checks/openmetrics/v2/scraper.py

```diff
@@ -0,0 +1,324 @@
+                    hostname = self.hostname_formatter(hostname)
+
+            self.submit_telemetry_number_of_processed_metric_samples()
+            yield sample, tags, hostname
+
+    def stream_connection_lines(self):
+        with self.get_connection() as connection:
+            for line in connection.iter_lines(chunk_size=self.STREAM_CHUNK_SIZE, decode_unicode=True):
+                yield line
+
+    def filter_connection_lines(self, line_streamer):
+        for line in line_streamer:
+            if self.raw_line_filter.search(line):
+                self.submit_telemetry_number_of_ignored_lines()
+            else:
+                yield line
+
+    def get_connection(self):
+        try:
+            response = self.send_request()
+        except Exception as e:
+            self.submit_health_check(ServiceCheck.CRITICAL, message=str(e))
+            raise
+        else:
+            try:
+                response.raise_for_status()
+            except Exception as e:
+                self.submit_health_check(ServiceCheck.CRITICAL, message=str(e))
+                response.close()
+                raise
+            else:
+                self.submit_health_check(ServiceCheck.OK)
+
+                # Never derive the encoding from the locale
+                if response.encoding is None:
+                    response.encoding = 'utf-8'
+
+                self.submit_telemetry_endpoint_response_size(response)
+                return response
+
+    def send_request(self, **kwargs):
+        kwargs['stream'] = True
+        return self.http.get(self.endpoint, **kwargs)
+
+    def set_dynamic_tags(self, *tags):
+        self.tags = tuple(chain(self.static_tags, tags))
+
+    def submit_health_check(self, status, **kwargs):
+        if self.enable_health_service_check:
+            self.service_check(self.SERVICE_CHECK_HEALTH, status, tags=self.static_tags, **kwargs)
+
+    def submit_telemetry_number_of_total_metric_samples(self, metric):
+        self.count('telemetry.metrics.input.count', len(metric.samples), tags=self.tags)
+
+    def submit_telemetry_number_of_ignored_metric_samples(self, metric):
+        self.count('telemetry.metrics.ignored.count', len(metric.samples), tags=self.tags)
+
+    def submit_telemetry_number_of_processed_metric_samples(self):
+        self.count('telemetry.metrics.processed.count', 1, tags=self.tags)
+
+    def submit_telemetry_number_of_ignored_lines(self):
+        self.count('telemetry.metrics.blacklist.count', 1, tags=self.tags)
+
+    def submit_telemetry_endpoint_response_size(self, response):
+        content_length = response.headers.get('Content-Length')
+        if content_length is not None:
+            content_length = int(content_length)
+        else:
+            content_length = len(response.content)
+
+        self.gauge('telemetry.payload.size', content_length, tags=self.tags)
+
+    def __getattr__(self, name):
+        # Forward all unknown attribute lookups to the check instance
```

Comment:
Why do we want to do this?  Seems like it would make it difficult to debug?

Reply:
So we can access the logger & send metrics for the telemetry methods